handshake.md

```diff
@@ -496,6 +502,13 @@ Reading a Cable Wire Protocol message MUST follow these steps:
 3. The resulting bytes `plaintext` may then be parsed as a Cable Wire Protocol
    message.
 
+### 5.5 End of stream
+After a complete Cable Wire Protocol message is sent, regardless of whether
+it consists of one or several fragments, the sender MUST write a message of
+`length=0`. This final message serves as an end of stream marker for the
+receiving peer and enables the peer to differentiate between an intentionally
+terminated stream and a dropped connection.
```

Comment:
```suggestion
### 5.5 End of stream
When a host has decided to terminate the exchange of messages, they MUST send a message of length zero to indicate this intention, and MUST NOT send any further messages.

The host receiving a message of length zero SHOULD respond by writing back a message of length zero, to indicate it has also finished writing. An implementation SHOULD have a time-out of some kind in case the other side does not transmit an end-of-stream marker or it is truncated by an attacker.

When a host has both sent and received a length-zero message, it is then safe for the underlying transport to do its own disconnect logic, if any.
```